File: factorial.cs
using System;

class Factorial
{
    static int FactorialCalc(int n)
    {
        if (n == 0)
            return 1;
        else
            return n * FactorialCalc(n - 1);
    }

    static void Main()
    {
        int num = 5;
        Console.WriteLine("The factorial of " + num + " is " + FactorialCalc(num));
    }
}

File: factorial.cpp
#include <iostream>

int factorial(int n) {
    if (n == 0)
        return 1;
    else
        return n * factorial(n - 1);
}

int main() {
    int num = 5;
    std::cout << "The factorial of " << num << " is " << factorial(num) << std::endl;
    return 0;
}
